public/app/features/search/service/frontend.ts

```diff
@@ -0,0 +1,102 @@
+import { DataFrameView, SelectableValue, ArrayVector } from '@grafana/data';
+import { TermCount } from 'app/core/components/TagFilter/TagFilter';
+
+import { DashboardQueryResult, GrafanaSearcher, QueryResponse, SearchQuery } from '.';
+
+export class FrontendSearcher implements GrafanaSearcher {
+  full = new Map<string, FullResultCache>();
+
+  constructor(private parent: GrafanaSearcher) {}
+
+  async search(query: SearchQuery): Promise<QueryResponse> {
+    if (query.facet?.length) {
+      throw new Error('facets not supported!');
+    }
+    // Don't bother... not needed for this exercise
+    if (query.tags?.length || query.ds_uid?.length) {
+      return this.parent.search(query);
+    }
+
+    // TODO -- make sure we refresh after a while
+    let dash = await this.getDashboards();
+    const view = dash.search(query.query);
+    return {
+      isItemLoaded: () => true,
+      loadMoreItems: async (startIndex: number, stopIndex: number): Promise<void> => {},
+      totalRows: view.length,
+      view,
+    };
+  }
+
+  async getDashboards(): Promise<FullResultCache> {
+    let res = this.full.get('dashboard');
+    if (res) {
+      return Promise.resolve(res);
+    }
+    const v = await this.parent.search({
+      kind: ['dashboard'],
+      limit: 5000, // max for now
+    });
+
+    res = new FullResultCache(v.view);
+    this.full.set('dashboard', res);
+    return res;
+  }
+
+  async starred(query: SearchQuery): Promise<QueryResponse> {
+    return this.parent.starred(query);
+  }
+
+  // returns the appropriate sorting options
+  async getSortOptions(): Promise<SelectableValue[]> {
+    return this.parent.getSortOptions();
+  }
+
+  async tags(query: SearchQuery): Promise<TermCount[]> {
+    return this.parent.tags(query);
+  }
+}
+
+class FullResultCache {
+  readonly lower: string[];
+  empty: DataFrameView<DashboardQueryResult>;
+
+  constructor(private full: DataFrameView<DashboardQueryResult>) {
```

Comment:
should we sort the dashboards alphabetically or is that part of @leeoniya's PR?

Reply:
the ranking pass will will be able to sort them alphabetically, so there's no need to do it in a prep step.

do we have any other attributes that should affect sorting? e.g. created, last modified, recently used, etc.?

Reply:
maybe tags?  the rest we will want to use for direct sorting, but not in this first round